docs/v8-migration-api.md

```diff
@@ -0,0 +1,392 @@
+
+  This change will affect your integration only when you are using `tokens: true` and are depending on the extra `tokens` AST output.
+
+  ```js title="babel-integration.js"
+  import { parse } from "@babel/parser";
+
+  const { tokens } = parse("class C { #priv }", { tokens: true });
+  tokens.filter(t => t.start >= 10 && t.end <= 15) // get tokens for `#priv`
+  // Babel 7
+  // [
+  //  Token (#) { value: "#", start: 10, end: 11 },
+  //  Token (name) { value: "priv", start: 11, end: 15 }
+  // ]
+  // Babel 8
+  // [
+  //  Token (privateName) { value: "priv", start: 10, end: 15 }
+  // ]
+  ```
+  **Migration**: Adapt to the new `privateName` token. If you want to restore to the Babel 7 behaviour, manually process the `privateName` token into the `#` token and the `name` token ([example](https://github.com/babel/babel/blob/7e60a93897f9a134506251ea51269faf4d02a86c/packages/babel-parser/src/parser/statement.ts#L86-L110)).
+
+- Tokenize string template as `templateNonTail` and `templateTail` ([#13919](https://github.com/babel/babel/pull/13919))
+
+  This change will affect your integration only when you are using `tokens: true` and are depending on the extra `tokens` AST output.
+
+  ```js title="babel-integration.js"
+  import { parse } from "@babel/parser";
+
+  const { tokens } = parse("`head${x}middle${y}tail`", { tokens: true });
+  console.log(tokens); // print tokens
+  // Babel 7
+  // [
+  //  Token (`),
+  //  Token (template) { value: "head" }, Token (${),
+  //  Token (name) { value: "x" }, Token (}),
+  //  Token (template) { value: "middle" }, Token (${),
+  //  Token (name) { value: "y" }, Token (}),
+  //  Token (template) { value: "tail" }
+  //  Token (`)
+  // ]
+  // Babel 8
+  // [
+  //  Token (templateNonTail) { value: "head" },
+  //  Token (name) { value: "x" },
+  //  Token (templateNonTail) { value: "middle" },
+  //  Token (name) { value: "y" },
+  //  Token (templateTail) { value: "tail" }
+  // ]
+  ```
+
+  **Migration**: Adapt to the new token design. If you want to restore to the Babel 7 behaviour, manually transform them to the Babel 7 tokens ([example](https://github.com/babel/babel/blob/7e60a93897f9a134506251ea51269faf4d02a86c/packages/babel-parser/src/parser/statement.ts#L116-L188)).
+
+### `@babel/traverse`
+
+![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+- [Allow skipped `NodePath`s to be requeued](https://github.com/babel/babel/blob/43b623c1f1e86e6fb86cae8d955a84fd924380a4/packages/babel-traverse/src/path/context.js#L241-L247) ([#13291](https://github.com/babel/babel/pull/13291))
+
+  **Notes**: `NodePath#requeue()` can requeue a skipped NodePath. This is actually a bugfix, but it causes an infinite loop in the tdz implementation of `@babel/plugin-transform-block-scoping` so we defer the change to Babel 8.
+
+![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
```

Comment:
We can probably avoid mentioning this, since the only breakage it causes is when used with old Babel 7 packages?

Reply:
If it breaks us, chances are it may break other plugins. Let's keep plugin authors informed that such changes may break their plugins. I will place this one after the `scope.rename`. Is that good?